Makefile

```diff
@@ -259,13 +259,22 @@ amazon-cluster-manifest:
 	@pach-deploy amazon $(BUCKET_NAME) $(AWS_ID) $(AWS_KEY) $(AWS_TOKEN) $(AWS_REGION) $(STORAGE_NAME) $(STORAGE_SIZE)
 
 amazon-cluster:
-	aws s3api create-bucket --bucket $(BUCKET_NAME) --region $(AWS_REGION)
+	aws s3api create-bucket --bucket $(BUCKET_NAME) --create-bucket-configuration LocationConstraint=$(AWS_REGION)
```

Comment:
@jdoliner  For your worry, I do some additional tests to justify my request. I used a new aws account to do the following experiments with the default-region us-west-2.
The result is here, and a screenshot is shown to describe the situation.
```
# aws s3api create-bucket --bucket shengjieTest1 --region us-west-2
An error occurred (IllegalLocationConstraintException) when calling the CreateBucket operation:
The unspecified location constraint is incompatible for the region specific endpoint this request was sent to.
# aws s3api create-bucket --bucket shengjieTest1 --region us-east-1
{
    "Location": "/shengjieTest1"
}
# aws s3api create-bucket --bucket shengjietest2 --create-bucket-configuration LocationConstraint=us-west-2
{
    "Location": "http://shengjietest2.s3.amazonaws.com/"
}
# aws s3api create-bucket --bucket shengjietest5 --region us-west-2
{
    "Location": "/shengjietest5"
}
# aws s3api create-bucket --bucket shengjieTest4 --create-bucket-configuration LocationConstraint=us-west-2
An error occurred (InvalidBucketName) when calling the CreateBucket operation: The specified bucket is not valid.
#aws s3api create-bucket --bucket shengjieTest4 --create-bucket-configuration LocationConstraint=us-east-1
An error occurred (IllegalLocationConstraintException) when calling the CreateBucket operation: The us-east-1 location constraint is incompatible for the region specific endpoint this request was sent to.
```
In conclusion, we summarize the error response here

1.  The s3-bucket name is allowed to involve the uppercase letter only if the location of s3-bucket is `us-east-1` with the `create-bucket region --bucket <name> --region <region_name> `
2.  If we establish the s3-bucket with a name involving the uppercase lestter

- 2.1 within the `--create-bucket-configuration <default-region>`, we have the error response as `InvalidBucketName` 
- 2.2 within the `--create-bucket-configuration <none-default-region>` we have the error response as `IllegalLocationConstraintException`
- 2.3 within the `--region <region_other _than_us-east-1>` we have the error response as `IllegalLocationConstraintException`
- 2.4 within the `--region <us-east-1>` , no error response, the bucket can be established.

If the bucket name follows the following points, than it can be used in both options.

- Bucket names should not contain upper-case letters
- Bucket names should not contain underscores (_)
- Bucket names should not end with a dash
- Bucket names should be between 3 and 63 characters long
- Bucket names cannot contain dashes next to periods (e.g., my-.bucket.com and my.-bucket are invalid)
- Bucket names cannot contain periods

Therefore, from my opinion, a troubleshooting issue is better than revising the code. And the original code don't need to be revised. Please focus on the second part `make amazon-clean`. Thanks.